packages/framework/ai-collab/src/explicit-strategy/utils.ts

```diff
@@ -0,0 +1,60 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/**
```

Comment:
Do we have anything tracking moving these utilities to a shared location? `fail` should probably live next to our `assert`. The map utilities could probably go in `core-utils`.

@noencke thoughts?

Reply:
`fail` is kind of in a weird place because it doesn't place nicely with our assert tagging. `fail` is supposed to merely be a convenient way to assert (do `x ?? fail()` rather than `assert(x !== undefined); x`) but it's not equivalent because we don't replace the message with a hex error string like we do with our other asserts. So, I wouldn't recommend promoting it to the framework level until we do the assert tagging (IIRC, Craig looked into it and there was some reason why it was difficult to do), and in the meantime I've stopped using `fail` very much even when it's convenient, because of the assert tagging problem.

A lot of the other utilities in here seem fair game to elevate though.